Master

filip-szczepankiewicz opened the pull request:
Updated dtd_gamma to support negative variance

filip-szczepankiewicz committed:
Updated file to use mdm_fn_nii2bvalbvec

filip-szczepankiewicz committed:
Update gamma model to support negative variance

To get distribution of variances symmetric expectancy near zero the
variances can now be negative. However, the signal is constrined to be
real.
Added repeated random guess to avoid extremely poor fitting results due
to very bad guesses.
Fixerd bugg so that random guess is updated between iterations
Updated guess in second fitting (if using weights)
Adde UT for negative variances

markus-nilsson commented:
Found two things when testing the code

1) Generated .nii uFA can become complex, which is not handled by many viewers. I think this needs to be managed, in particular since uFA is a "display parameter" and not a model parameter as such

2) It would be good and convenient to have the random guess initialization implemented as a msf_fit_something function.

filip-szczepankiewicz committed:
Merge remote-tracking branch 'refs/remotes/markus-nilsson/master'

filip-szczepankiewicz committed:
Merge remote-tracking branch 'refs/remotes/markus-nilsson/master'

filip-szczepankiewicz committed:
Merge remote-tracking branch 'refs/remotes/markus-nilsson/master'

filip-szczepankiewicz committed:
UT for dtd_gamma that uses negative variance

Unit test for dtd_gamma was updated with a test where model assumes
negative variance.

filip-szczepankiewicz committed:
Added msf_fit_random_guess and updated dtd_gamma

msf_fit_random_guess provides multiple random guesses where the one with
least residual is returned. dtd_gamma is updated to use this function.

filip-szczepankiewicz committed:
Dissalow complex uFA values

Due to the possibility of negative variances, the uFA sometimes became
complex. This is now fixed by storing only the real component of uFA.

filip-szczepankiewicz committed:
Textfix

Something bugged with GitHub. Text got cut off.

filip-szczepankiewicz committed:
Fixed bugg in powder averaging

Fixed bug where opt was not properly handled to both image and xps
powder-averaging.
Updated tolerance for b-values in unique search to 1 s/mm2. The
tolerance of 200 s/mm2 is much to high for proper use.
